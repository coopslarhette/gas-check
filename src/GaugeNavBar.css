.GaugeNavBar {
    border-radius: 8px;
    background-color: #53afed;
}

.github-logo {
    max-height: 40px;
}
