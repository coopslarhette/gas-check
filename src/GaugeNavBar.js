import React from 'react'
import './GaugeNavBar.css'
import Navbar from 'react-bootstrap/Navbar'
import NavbarBrand from 'react-bootstrap/NavbarBrand'

function GaugeNavBar() {
  return (
    <div className="GaugeNavBar">
      <Navbar>
        <NavbarBrand href="https://coopslarhette.github.io/gauge/">
          Gauge
        </NavbarBrand>
        <div className="ml-auto">
          <a href="https://github.com/coopslarhette">
            <img alt={'github page'} src="../public/github-logo.png"/>
          </a>
        </div>
      </Navbar>
    </div>
  )
}

export default GaugeNavBar
