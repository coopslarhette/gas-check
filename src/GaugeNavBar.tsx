import React from 'react'
import './GaugeNavBar.css'
import Navbar from 'react-bootstrap/Navbar'
import NavbarBrand from 'react-bootstrap/NavbarBrand'
import GitHubLogo from './img/github-logo.png'

function GaugeNavBar(): JSX.Element {
  return (
    <div className="GaugeNavBar">
      <Navbar>
        <NavbarBrand href="https://coopslarhette.github.io/gauge/">
          Gauge
        </NavbarBrand>
        <div className="ml-auto">
          <a href="https://github.com/coopslarhette">
            <img className="github-logo" alt="github page" src={GitHubLogo} />
          </a>
        </div>
      </Navbar>
    </div>
  )
}

export default GaugeNavBar
