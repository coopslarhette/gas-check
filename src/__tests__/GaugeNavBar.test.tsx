import React from 'react'
import ReactDOM, { unmountComponentAtNode } from 'react-dom'
import renderer from 'react-test-renderer'
import GaugeNavBar from '../GaugeNavBar'

describe('Nav bar component', () => {
  let div: HTMLDivElement

  beforeEach(() => {
    div = document.createElement('div')
    document.body.appendChild(div)
  })

  afterEach(() => {
    unmountComponentAtNode(div)
    div.remove()
  })

  it('renders without crashing', () => {
    ReactDOM.render(<GaugeNavBar />, div)
  })

  it('renders all parts correctly', () => {
    const tree = renderer
      .create(<GaugeNavBar />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
