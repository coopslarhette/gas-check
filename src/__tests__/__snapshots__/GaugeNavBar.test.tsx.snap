// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar component renders all parts correctly 1`] = `
<div
  className="GaugeNavBar"
>
  <nav
    className="navbar navbar-expand navbar-light"
  >
    <a
      className="navbar-brand"
      href="https://coopslarhette.github.io/gauge/"
    >
      Gauge
    </a>
    <div
      className="ml-auto"
    >
      <a
        href="https://github.com/coopslarhette"
      >
        <img
          alt="github page"
          className="github-logo"
          src="github-logo.png"
        />
      </a>
    </div>
  </nav>
</div>
`;
